src/lib/select/select.spec.ts

```diff
@@ -62,3203 +70,3284 @@ describe('MatSelect', () => {
+        expect(optionInstances[2].selected).toBe(false);
+      }));
 
-      fixture.detectChanges();
-      fixture.debugElement.query(By.css('.mat-select-trigger')).nativeElement.click();
-      fixture.detectChanges();
-      flush();
+      it('should deselect other options when one is programmatically selected', fakeAsync(() => {
+        let control = fixture.componentInstance.control;
+        let foods = fixture.componentInstance.foods;
 
-      (overlayContainerElement.querySelector('mat-option') as HTMLElement).click();
-      fixture.detectChanges();
-      flush();
+        trigger.click();
+        fixture.detectChanges();
+        flush();
 
-      const select = fixture.debugElement.nativeElement.querySelector('mat-select');
+        let options =
+            overlayContainerElement.querySelectorAll('mat-option') as NodeListOf<HTMLElement>;
 
-      expect(document.activeElement).toBe(select, 'Expected trigger to be focused.');
-    }));
-  });
+        options[0].click();
+        fixture.detectChanges();
+        flush();
 
-  describe('disabled behavior', () => {
-    it('should disable itself when control is disabled programmatically', fakeAsync(() => {
-      const fixture = TestBed.createComponent(BasicSelect);
-      fixture.detectChanges();
+        control.setValue(foods[1].value);
+        fixture.detectChanges();
 
-      fixture.componentInstance.control.disable();
-      fixture.detectChanges();
-      let trigger =
-          fixture.debugElement.query(By.css('.mat-select-trigger')).nativeElement;
-      expect(getComputedStyle(trigger).getPropertyValue('cursor'))
-          .toEqual('default', `Expected cursor to be default arrow on disabled control.`);
+        trigger.click();
+        fixture.detectChanges();
+        flush();
 
-      trigger.click();
-      fixture.detectChanges();
+        options = overlayContainerElement.querySelectorAll('mat-option') as NodeListOf<HTMLElement>;
 
-      expect(overlayContainerElement.textContent)
-          .toEqual('', `Expected select panel to stay closed.`);
-      expect(fixture.componentInstance.select.panelOpen)
-          .toBe(false, `Expected select panelOpen property to stay false.`);
+        expect(options[0].classList)
+            .not.toContain('mat-selected', 'Expected first option to no longer be selected');
+        expect(options[1].classList)
+            .toContain('mat-selected', 'Expected second option to be selected');
 
-      fixture.componentInstance.control.enable();
-      fixture.detectChanges();
-      expect(getComputedStyle(trigger).getPropertyValue('cursor'))
-          .toEqual('pointer', `Expected cursor to be a pointer on enabled control.`);
+        const optionInstances = fixture.componentInstance.options.toArray();
 
-      trigger.click();
-      fixture.detectChanges();
+        expect(optionInstances[0].selected)
+            .toBe(false, 'Expected first option to no longer be 
```

Comment:
Since it's a single test it doesn't need the `beforeEach`, it could be moved outside or into the "special cases".